apps/api/src/app/environments/environments.bridge.controller.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  Body,
+  ClassSerializerInterceptor,
+  Controller,
+  Get,
+  NotFoundException,
+  Options,
+  Param,
+  Post,
+  Query,
+  Req,
+  Res,
+  UseInterceptors,
+} from '@nestjs/common';
+import { Request, Response } from 'express';
+import { StepTypeEnum, WorkflowTypeEnum } from '@novu/shared';
+import { ApiTags } from '@nestjs/swagger';
+import { decryptApiKey } from '@novu/application-generic';
+import { EnvironmentRepository, NotificationTemplateEntity, NotificationTemplateRepository } from '@novu/dal';
+import { Client, Event, workflow, Step, Workflow } from '@novu/framework';
+import { ApiCommonResponses } from '../shared/framework/response.decorator';
+import { NovuNestjsHandler } from './novu-nestjs-handler';
+
+// Unfortunately we need this mapper because the `in_app` step type uses `step.inApp()` in Framework.
+const stepFnFromStepType: Record<Exclude<StepTypeEnum, StepTypeEnum.CUSTOM | StepTypeEnum.TRIGGER>, keyof Step> = {
+  [StepTypeEnum.IN_APP]: 'inApp',
+  [StepTypeEnum.EMAIL]: 'email',
+  [StepTypeEnum.SMS]: 'sms',
+  [StepTypeEnum.CHAT]: 'chat',
+  [StepTypeEnum.PUSH]: 'push',
+  [StepTypeEnum.DIGEST]: 'digest',
+  [StepTypeEnum.DELAY]: 'delay',
+};
+
+@ApiCommonResponses()
+@Controller('/environments')
+@UseInterceptors(ClassSerializerInterceptor)
+@ApiTags('Environments')
+export class EnvironmentsBridgeController {
+  constructor(
+    private workflowsRepository: NotificationTemplateRepository,
+    private environmentsRepository: EnvironmentRepository
+  ) {}
+
+  @Get('/:environmentId/bridge')
+  async getHandler(@Req() req: Request, @Res() res: Response, @Param('environmentId') environmentId: string) {
+    /*
+     * TODO: remove the GET handler, it's only used right now for the bridge health check,
+     * and it's not needed for UI-defined workflows.
+     *
+     * I'm leaving it here as a reminder that we need to remove the UI health-check for Novu-managed
+     * bridge environments.
+     */
+    const novuBridgeHandler = new NovuNestjsHandler({
+      workflows: [],
+      client: new Client({
+        strictAuthentication: false,
+        secretKey: await this.getApiKey(environmentId),
+      }),
+    });
+    await novuBridgeHandler.handleRequest(req, res, 'GET');
+  }
+
+  @Post('/:environmentId/bridge')
+  async postHandler(
+    @Req() req: Request,
+    @Res() res: Response,
+    @Param('environmentId') environmentId: string,
+    @Query('workflowId') workflowId: string,
+    @Body() event: Event
+  ) {
+    const foundWorkflow = await this.getWorkflow(environmentId, workflowId);
+
+    const programmaticallyCreatedWorkflow = this.createWorkflow(foundWorkflow, event.controls);
+
+    const novuBridgeHandler = new NovuNestjsHandler({
+      workflows: [programmaticallyCreatedWorkflow],
+      client: new Client({ strictAuthentication: true, secretKey: await this.getApiKey(environmentId) }),
+    });
+
+    await novuBridgeHandler.handleRequest(req, res, 'POST');
```

Comment:
Would it be possible to enhance the clarity of this code by adding a JavaScript comment to explain the functionality of this part and its origin of invocation? It seems that this section is not self-explanatory.

Reply:
I have added some comments to describe the control flow. Let me know if that helps with describing the code.